dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/deletion/JdbcDeletionHandler.java

```diff
@@ -63,4 +64,14 @@ protected final int delete( String sql, Map<String, Object> parameters )
     {
         return npTemplate.update( sql, parameters );
     }
+
+    protected final String firstMatch( String sql, Map<String, Object> parameters )
+    {
+        if ( !sql.toLowerCase().contains( "limit 1" ) )
```

Comment:
A minor improvement here would be to do a regex check for `limit [0-9]+` as in theory the SQL could contain a limit clause for another limit (e.g. 10), and if appending another limit clause, it would crash.